include/electionguard/elgamal.hpp

```diff
@@ -351,39 +345,66 @@ namespace electionguard
     /// This value indicates the maximum length of the plaintext that may be
     /// encrypted. The padding scheme applies two bytes for length of padding
-    /// plus padding bytes. If padding is not to be applied then the
-    /// max_len parameter must be NO_PADDING. and the plaintext must
-    /// be a multiple of the block length (32) and the ciphertext will be
-    /// the same size. If the max_len is set to  something other than
-    /// NO_PADDING and the allow_truncation parameter is set to
+    /// plus padding bytes.
+    ///
+    /// If allow_truncation parameter is set to
     /// true then if the message parameter data is longer than
-    /// max_len then it will be truncated to max_len. If the max_len is set to
-    /// something other than NO_PADDING and the allow_truncation parameter
+    /// max_len then it will be truncated to max_len.
+    /// If the allow_truncation parameter
     /// is set to false then if the message parameter data is longer than
     /// max_len then an exception will be thrown.
     ///
-    /// <param name="message"> Message to hashed elgamal encrypt. </param>
-    /// <param name="nonce"> Randomly chosen nonce in [1,Q). </param>
-    /// <param name="publicKey"> ElGamal public key. </param>
-    /// <param name="descriptionHash"> Hash of the ballot description. </param>
-    /// <param name="max_len"> If padding is to be applied then this indicates the
-    ///  maximum length of plaintext, must be one padded_data_size_t enumeration
-    ///  values. If padding is not to be applied then this parameter must use
-    ///  the NO_PADDING padded_data_size_t enumeration value.</param>
-    /// <param name="allow_truncation"> Truncates data to the max_len if set to
-    /// true. If max_len is set to NO_PADDING then this parameter is ignored. </param>
+    /// <param name="message">Message to hashed elgamal encrypt.</param>
+    /// <param name="nonce">Randomly chosen nonce in [1,Q).</param>
+    /// <param name="hashPrefix">A prefix value for the hash used to create the session key.</param>
+    /// <param name="publicKey">ElGamal public key.</param>
+    /// <param name="seed">Hash of the ballot description.</param>
+    /// <param name="max_len">Indicates the maximum length of plaintext,
+    ///                       must be one of the `HASHED_CIPHERTEXT_PADDED_DATA_SIZE`
+    ///                       enumeration values.
+    /// </param>
+    /// <param name="allow_truncation">Truncates data to the max_len if set to true.
+    /// </param>
+    /// <param name="shouldUsePrecomputedValues">If true, the function will attempt
+    ///                                          to use a precomputed value form the precompute buffer
+    /// </param>
+    /// <returns>A ciphertext triple.</returns>
+    /// </summary>
+    EG_API std::unique_ptr<HashedElGamalCiphertext>
+    hashedElgamalEncrypt(std::vector<uint8_t> message, const ElementModQ &nonce,
```

Comment:
so now there are 2 overrides. one accepts a max size and can handle padding and truncation, and the other does not accept those parameters and forces the data to be specifically shaped. this gets rid of the `NO_PADDING` enum value and moves that complexity behind the api surface